### DISCUSSION TITLE
What type of plugins to use?

### DISCUSSION THREAD

**Discussion by fojerhar** [apohl79/audiogridder#599] [General]
Hi
If I use AG in pro tools, does that matter, if I use AU or VST or VST3 inside AG? 
Can I mix them in a chain? (I know that I can, but it is healthier to use one type, or that is equal programming and protocol wise?)
And one main thing? Who does the cpu ditribution, or is it a big magic, to solve this like you did? Because in PT 2 or 4 cores are working only if I use plugs only inside, but here voilá, can be solved and the magic happened? The daws why cannot do this? (this is a happy complaining)

**Comment by apohl79** [comment]
> does that matter, if I use AU or VST or VST3 inside AG?

Generally, from Pro Tools no, as AAX supports only 32bit floats (same as AU). Only VST can do 64bit. So no degradation can happen.

Some plugins work only in one specific format in AG, so you have no choice. If you have a choice and all formats work, VST3 is probably the best, as it supports presets offered by the plugin vendor and works on all platforms.

> Can I mix them in a chain? (I know that I can, but it is healthier to use one type, or that is equal programming and protocol wise?)

You can mix them without any downsides.

> And one main thing? Who does the cpu ditribution, or is it a big magic, to solve this like you did? Because in PT 2 or 4 cores are working only if I use plugs only inside, but here voilá, can be solved and the magic happened? The daws why cannot do this? (this is a happy complaining)

AG runs each plugin chain (each AG plugin you add in PT runs a "chain" on the AG server) on a separate thread or process (with sandboxing). This does the CPU distribution. No magic. I can't say anything about PT's internal design, but it is likely grown over the years. AG compared to this is very simple and can do just that without side effects.

**Reply by fojerhar** [reply]
Thank you for the proper answer

**Reply by lesbrunn** [reply]
> AG runs each plugin chain (each AG plugin you add in PT runs a "chain" on the AG server) on a separate thread or process (with sandboxing). This does the CPU distribution. No magic. I can't say anything about PT's internal design, but it is likely grown over the years. AG compared to this is very simple and can do just that without side effects.

Hi, that sounds good. Unfortunately, I haven't had very happy results. CPU usage seems to seep into PT, causing severe spikes in PT that make work difficult if not impossible. What can I do about this?

**Reply by giacometti777** [reply]
> > AG runs each plugin chain (each AG plugin you add in PT runs a "chain" on the AG server) on a separate thread or process (with sandboxing). This does the CPU distribution. No magic. I can't say anything about PT's internal design, but it is likely grown over the years. AG compared to this is very simple and can do just that without side effects.
> 
> Hi, that sounds good. Unfortunately, I haven't had very happy results. CPU usage seems to seep into PT, causing severe spikes in PT that make work difficult if not impossible. What can I do about this?

Make sure you are using a wired connection AND turn off wifi on any slave machines. Wifi causes CPU spikes in my experience and your machine running the DAW will sometimes use the Wifi connection with the slave through the router causing spikes.

**Reply by lesbrunn** [reply]
> Make sure you are using a wired connection AND turn off wifi on any slave machines. Wifi causes CPU spikes in my experience and your machine running the DAW will sometimes use the Wifi connection with the slave through the router causing spikes.

Thanks! I'm using a wired connection. I'm still experimenting with various configs... and I know some people here are getting good results.
